lib/waterfoul/mmu.rb

```diff
@@ -33,26 +30,14 @@ def initialize
     def [](i)
       raise MemoryOutOfBounds if i > MEMORY_SIZE || i < 0
 
-      case i
-      when 0xFF00
+      if i == 0xFF00
         Input.read_keyboard @memory[i]
-      when 0x0000...0x8000 # ROM Bank 0 + n
-        # if the boot rom is enabled and the address is < 0x100
-        if @map_boot_rom && i <= BOOT_ROM_END_MEM_LOC
-          BootROM[i]
-        else
-          @cartridge[i]
-        end
-      when 0x8000...0xA000 # Video RAM
+      elsif i >= 0x8000 && i < 0xA000 or i >= 0xC000 && i < 0xE000 or i >= 0xFE00
```

Comment:
This is less nice than the ranges, but avoiding their allocation is a major speedup on the standard ruby/MRI.